infrastructure/iac/iac.go

```diff
@@ -73,33 +78,64 @@ func (iac *Scanner) SupportedCommands() []snyk.CommandName {
 func (iac *Scanner) Scan(ctx context.Context, path string, _ string) (issues []snyk.Issue) {
 	if ctx.Err() != nil {
 		log.Info().Msg("Cancelling IAC scan - IAC scanner received cancellation signal")
-		return []snyk.Issue{}
+		return
 	}
 
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
 	documentURI := uri.PathToUri(path) // todo get rid of lsp dep
 	if !iac.isSupported(documentURI) {
 		return
 	}
-	p := progress.NewTracker(false)
+	p := progress.NewTracker(false) // todo - get progress trackers via DI
 	p.BeginUnquantifiableLength("Scanning for Snyk IaC issues", path)
-	defer p.End("Snyk Iac Scan completed.")
 
 	var workspacePath string
 	if uri.IsDirectory(documentURI) {
 		workspacePath = uri.PathFromUri(documentURI)
 	} else {
 		workspacePath = filepath.Dir(uri.PathFromUri(documentURI))
 	}
+	iac.mutex.Lock()
```

Comment:
If the previous scan is running, how can the new scan relock the mutex here and cancel it?

Reply:
Hm... I see, the runningScans map will do that.